### PR TITLE
Added a trait for expression creation methods.

### PR DESCRIPTION
**Pull request by fubhy**

### CODE REVIEW

File: src/Tests/RulesEngineTest.php [hunk 1]
```diff
@@ -29,18 +29,31 @@ public static function getInfo() {
...
-        ->addCondition($this->createExpression('rules_and')
-          ->addCondition($this->createCondition('rules_test_false'))
-          ->addCondition($this->createCondition('rules_test_true')->negate())
-          ->negate()));
-    $rule->addAction($this->createAction('rules_test_log'));
+    // Create an 'and' condition container and add conditions to it.
+    $and = $this->createRulesAnd()
+      ->addCondition($this->createRulesCondition('rules_test_false'))
+      ->addCondition($this->createRulesCondition('rules_test_true')->negate())
+      ->negate();
```
Comment (klausi) [inline src/Tests/RulesEngineTest.php:36]: Could you also include the result of the and in the comment like "The 'and' evaluates to TRUE"?

File: src/Tests/RulesEngineTest.php [hunk 2]
```diff
@@ -29,18 +29,31 @@ public static function getInfo() {
...
+
+    // Create an 'or' condition container and add conditions to it, including
+    // the previously created 'and' condition container.
+    $or = $this->createRulesOr()
+      ->addCondition($this->createRulesCondition('rules_test_true')->negate())
+      ->addCondition($this->createRulesCondition('rules_test_false'))
+      ->addCondition($and);
```
Comment (klausi) [inline src/Tests/RulesEngineTest.php:46]: Same here, it would be nice to know the intermediate outcome of this OR in the comment.

**Comment by fubhy** [comment]
Regarding your comments... What about we execute these intermediate steps too to show that? Like... add assertions for those?

**Comment by klausi** [comment]
Sure, that would make it easier to debug this test and show at what step it went wrong. Good idea!

**Comment by fubhy** [comment]
Done.

**Pull request merged**